xm-commons-domainevent/src/main/java/com/icthh/xm/commons/domainevent/config/WebApiSource.java

```diff
@@ -0,0 +1,118 @@
+package com.icthh.xm.commons.domainevent.config;
+
+import com.icthh.xm.commons.domainevent.domain.DomainEvent;
+import com.icthh.xm.commons.domainevent.domain.HttpDomainEventPayload;
+import com.icthh.xm.commons.domainevent.domain.enums.DefaultDomainEventSource;
+import com.icthh.xm.commons.domainevent.service.EventPublisher;
+import com.icthh.xm.commons.logging.util.MdcUtils;
+import com.icthh.xm.commons.security.XmAuthenticationContext;
+import com.icthh.xm.commons.security.internal.SpringSecurityXmAuthenticationContextHolder;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+import org.springframework.web.util.ContentCachingResponseWrapper;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.net.http.HttpHeaders;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class WebApiSource extends HandlerInterceptorAdapter {
+
+    private static final String HEADER_TENANT = "x-tenant";
+
+    private final EventPublisher eventPublisher;
+    private final SpringSecurityXmAuthenticationContextHolder springSecurityXmAuthenticationContextHolder;
+
+    @Override
+    public void afterCompletion(HttpServletRequest request, HttpServletResponse response, Object handler, Exception ex) {
+
+        if (isIgnoredRequest(request, response)) {
+            return;
+        }
+
+        XmAuthenticationContext auth = springSecurityXmAuthenticationContextHolder.getContext();
+        if (auth == null || auth.isAnonymous()) {
+            String tenant = request.getHeader(HEADER_TENANT);
+            publishEvent(request, response, tenant, null, null);
+        } else {
+            String userLogin = auth.getLogin().orElse("");
+            String userKey = auth.getUserKey().orElse("");
+            String tenant = auth.getTenantName().orElse("");
+
+            publishEvent(request, response, tenant, userLogin, userKey);
+        }
+    }
+
+    // TODO
+    private boolean isIgnoredRequest(HttpServletRequest request, HttpServletResponse response) {
+        return false;
+    }
+
+    private void publishEvent(HttpServletRequest request, HttpServletResponse response, String tenant, String userLogin, String userKey) {
+        DomainEvent domainEvent = DomainEvent.builder()
```

Comment:
more fields should be initialized (see old implementation here `com.icthh.xm.commons.timeline.TimelineEventProducer#createEventJson` but consider refactoring if need):

`txId` - from `MdcUtils.getRid()`
`msName` - from `${spring.application.name}`
`operation` - please check the requirements for `Custom operation support` here: https://docs.google.com/document/d/1dAAtXo90hswG9_tVaf4BcUEOC7QQ0tt4ojYwIcwLUXM/edit#heading=h.qbczcsrb50f6
`source` - DefaultDomainEventSource.HTTP
`clientId` - from token claim `client_id`, more details are here (note in the document field clientId is referred as ctannelType): https://docs.google.com/document/d/1dAAtXo90hswG9_tVaf4BcUEOC7QQ0tt4ojYwIcwLUXM/edit#heading=h.ct5745avt4su
`aggregateId` -  try to resolve it as in this method: com.icthh.xm.commons.timeline.TimelineEventProducer#getEntityField (or propose better solution)
`aggregateType` - try to resolve it as in this method: com.icthh.xm.commons.timeline.TimelineEventProducer#getEntityField (or propose better solution)